pkg/codegen/go/gen_program_expressions.go

```diff
@@ -947,7 +947,7 @@ func (g *generator) argumentTypeName(destType model.Type, isInput bool) (result
 			}
 		}
 
-		if elmType != nil {
+		if elmType != nil && elmType != model.NoneType {
```

Comment:
Without this go codegen would write a type `[]` that is a slice without the type inside it.